plugins/moduleLoader.js

```diff
@@ -2,89 +2,125 @@
  * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
  * See LICENSE for license details.
  */
+/* eslint-disable global-require */
+/* eslint-disable import/no-dynamic-require */
 import Vue from 'vue'; // eslint-disable-line import/no-extraneous-dependencies
-// import { modules } from '@Root/modules.config';
+import additionalConfig from '@Root/modules.config'; // User modules configuration file
 
-const modules = [];
+class ModuleLoader {
+    constructor() {
+        this.config = this.setModulesConfiguration();
+        this.pagesConfig = this.setPagesConfiguration(this.config);
+        this.componentsConfig = this.setComponentsConfiguration(this.config);
+    }
+
+    get getPagesConfig() {
+        return this.pagesConfig;
+    }
+
+    get getComponentsConfig() {
+        return this.componentsConfig;
+    }
 
-const ModuleLoader = {
     install() {
-        this.getData().then((components) => {
-            components.forEach((component) => {
-                this.getComponent(component.path, component.module);
-            });
-        });
-    },
-    getRouterConfigs() {
-        return modules.reduce((previousObject, currentObject) => {
+        const { router, menu } = this.pagesConfig;
+        Vue.prototype.$modulesConfiguration = {
+            router,
+            menu,
+        };
+    }
+
+    requireForce(modulePath) { // force require
+        try {
+            return require(`${modulePath}`);
+        } catch (e) {
+            return false;
+        }
+    }
+
+    setModulesConfiguration() {
+        let summaryConfig = { pages: [], components: [] };
+        const {
+            pages: additionalConfigPages,
+            components: additionalConfigComponents,
+        } = additionalConfig;
```

Comment:
I think you should pass additionalConfig to class contructor instead of using it as variable in parent scope